buildpack/telemetry/dynatrace.py

```diff
@@ -5,23 +5,165 @@
 """
 import logging
 import os
+import json
 from urllib.parse import urljoin
 
-INGEST_ENDPOINT = "/api/v2/metrics/ingest"
+INGEST_ENDPOINT = "api/v2/metrics/ingest"
 
 
-def is_enabled():
+from buildpack import util
+# Environment variables for Dynatrace OneAgent
+# Only passed to the agent if set as environment variable
+default_env = {
+    # -- Environment variables for the integration
+    # "DT_PAAS_TOKEN": required, also used for telegraf integration
+    # "DT_SAAS_URL": required, also used for telegraf integration
+    "DT_TENANT": None,  # required for agent integration, dynatrace environment ID
+    "DT_TENANTTOKEN": None,  # optional, default value is get from manifest.json which is downloaded along with the agent installer
+    # -- Environment variables for orchestration
+    "DT_LOCALTOVIRTUALHOSTNAME": None,  # optional, default not set
+    "DT_APPLICATIONID": None,  # optional, default not set
+    "DT_NODE_ID": None,  # optional, default not set
+    "DT_CLUSTER_ID": None,  # optional, default not set
+    "DT_TAGS": None,  # optional tags e.g. MikesStuff easyTravel=Mike
+    "DT_CUSTOM_PROP": None,  # optional metadata e.g. Department=Acceptance Stage=Sprint
+    # -- Environment variables for troubleshooting
+    "DT_LOGSTREAM": "stdout",  # optional
+    "DT_LOGLEVELCON": None,  # Use this environment variable to define the console log level. Valid options are: NONE, SEVERE, and INFO.
+    "DT_AGENTACTIVE": None,  # Set to true or false to enable or disable OneAgent.
+    # -- Networking environment variables
+    "DT_NETWORK_ZONE": None,  # optional, Specifies to use a network zone. For more information
+    "DT_PROXY": None, # Optional, When using a proxy, use this environment variable to pass proxy credentials.
+}
+
+
+def stage(buildpack_dir, build_path, cache_path):
+    """
+    Downloads and unzips necessary OneAgent components
+    """
+    if is_agent_enabled():
+        try:
+            util.resolve_dependency(
+                "dynatrace.agent",
+                build_path,  # DOT_LOCAL_LOCATION,
+                buildpack_dir=buildpack_dir,
+                cache_dir=cache_path,  # CACHE_DIR,
+                unpack=True,
+                overrides={
+                    "url": os.environ.get("DT_SAAS_URL"),
+                    "environment": os.environ.get("DT_TENANT"),
+                    "token": os.environ.get("DT_PAAS_TOKEN"),
+                },
+                ignore_cache=True
```

Comment:
Can we check if the caching works now? Or please add a comment on why we are disabling this.

Reply:
added a comment for that. This needs to be investigated separately.